File: Coding Challenge 05/nearest_larger.m
function result=nearest_larger(array,i)
    diff = 1;
    while 1
        left = i - diff;
        right = i + diff;
        
        if (left > 0) && (array(left) > array(i))
            result = left;
            return
        elseif (right <= length(array)) && (array(right) > array(i))
            result = right;
            return
        elseif (left < 1) && (right > length(array))
            result = [];
            return
        end
        
        diff = diff + 1;
    end
end

File: Coding Challenge 05/nearest_larger.rb
#example input: nearest_larger([2,3,4,8], 2)
def nearest_larger(array, idx)
  diff = 1
  loop do
    left = idx - diff
    right = idx + diff

    if (left >= 0) && (array[left] > array[idx])
      return left
    elsif (right < array.length) && (array[right] > array[idx])
      return right
    elsif (left < 0) && (right >= array.length)
      return nil
    end

    diff += 1
  end
end
